src/modules/fancyzones/lib/ZoneWindow.cpp

```diff
@@ -535,6 +537,25 @@ ZoneWindow::MoveWindowIntoZoneByIndexSet(HWND window, const std::vector<int>& in
     }
 }
 
+IFACEMETHODIMP_(bool)
+ZoneWindow::SwitchFocusByDirection(HWND window, DWORD vkCode, bool cycle) noexcept
+{
+    std::shared_lock lock(m_mutex);
+    auto thisWindow = m_window.get();
+    auto activeZoneSet = m_activeZoneSet;
+    lock.unlock();
+
+    if (activeZoneSet)
+    {
+        if (activeZoneSet->SwitchFocusByDirection(window, thisWindow, vkCode, cycle))
+        {
+            SaveWindowProcessToZoneIndex(window);

```

Comment:
Not sure what this does exactly

Reply:
When a window is moved to a different zone, that zone is added to app zone history.